src/os_posix/os_alloc.c

```diff
@@ -134,6 +134,13 @@ __wt_realloc_aligned(WT_SESSION_IMPL *session,
 		WT_ASSERT(session, bytes_to_allocate != 0);
 		WT_ASSERT(session, bytes_allocated < bytes_to_allocate);
 
+		/*
+		 * We are going to allocate an aligned buffer: round up to
+		 * the aligned size.
+		 */
+		bytes_to_allocate =
+		    WT_ALIGN(bytes_to_allocate, S2C(session)->buffer_alignment);
+
```

Comment:
@agorrod, this looks questionable to me: we don't need to increase the size of the buffer (necessarily) to reach a particular alignment. I agree that means we won't know the size of the buffer with as much accuracy, but it also means we don't waste so much space?